tdp/core/collection.py

```diff
@@ -37,128 +37,74 @@ class Collection:
-            mandatory_path = path / mandatory_directory
-            if not mandatory_path.exists() or not mandatory_path.is_dir():
-                raise MissingMandatoryDirectoryError(
-                    f"{path} does not contain the mandatory directory {mandatory_directory}",
-                )
-        return Collection(path)
-
-    @property
-    def path(self) -> Path:
-        """Path to the collection."""
-        return self._path
+        self._services: Optional[List[Path]] = None
 
     @property
     def name(self) -> str:
         """Name of the collection."""
         return self._path.name
 
-    @property
-    def dag_directory(self) -> Path:
-        """Path to the DAG directory."""
-        return self._path / DAG_DIRECTORY_NAME
-
     @property
     def default_vars_directory(self) -> Path:
         """Path to the default variables directory."""
-        return self._path / DEFAULT_VARS_DIRECTORY_NAME
-
-    @property
-    def operations_directory(self) -> Path:
-        """Path to the operations directory."""
-        return self._path / OPERATION_DIRECTORY_NAME
-
-    @property
-    def schema_directory(self) -> Path:
-        """Path to the schema variables directory."""
-        return self._path / SCHEMA_VARS_DIRECTORY_NAME
+        return self._get_directory(DEFAULT_VARS_DIRECTORY_NAME)
 
     @property
     def dag_yamls(self) -> List[Path]:
         """List of DAG files in the YAML format."""
         if not self._dag_yamls:
-            self._dag_yamls = list(self.dag_directory.glob("*" + YML_EXTENSION))
+            self._dag_yamls = list(
+                self._get_directory(DAG_DIRECTORY_NAME).glob("*" + YML_EXTENSION)
+            )
         return self._dag_yamls
 
     @property
-    def operations(self) -> Dict[str, Path]:
-        """Dictionary of operations (playbooks) by name."""
+    def playbooks(self) -> Dict[str, Path]:
+        """Dictionary of playbooks."""
         if not self._operations:
             self._operations = {
                 playbook.stem: playbook
-                for playbook in self.operations_directory.glob("*" + YML_EXTENSION)
+                for playbook in self._get_directory(PLAYBOOKS_DIRECTORY_NAME).glob(
+                    "*" + YML_EXTENSION
+                )
             }
         return self._operations
 
-    def get_service_default_vars(self, service_name: str) -> List[Path]:
-        """Get the default variables for a service.
-
-        Args:
-            service_name: The name of the service.
-
-        Returns:
-            A list of tuples (name, path) of the default variables.
-        """
-        service_path = self.default_vars_directory / service_name
-        if not service_path.exists():
-            return []
-        return [(path.name, path) for path in service_path.glob("*" + YML_EXTENSION)]
-
     def get_service_schema(self, service_name: str) -> Dict:
-        """Get the schema of the service.
-
-        Args:
```

Comment:
Why removing `Args` and `Returns` doc ?

Reply:
They don't provide much value, but it'll better to fix them later along with doc refactoring.